### PR TITLE
Merge openshift/installer for 4.1.0-rc.3

### PR DESCRIPTION
**Pull request by stbenjam**
See also https://github.com/openshift-metal3/dev-scripts/pull/523

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/628/

**Comment by hardys** [comment]
Hmm the CI job failed because the branch doesn't rebase cleanly on master, and also if you try to build the branch without rebasing I see this error:

```
pkg/destroy/baremetal/baremetal.go:24:12: undefined: "github.com/openshift-metalkube/kni-installer/vendor/github.com/libvirt/libvirt-go".NewConnect
```

Which is weird because it's defined at https://github.com/libvirt/libvirt-go/blob/master/connect.go#L321 and that's in the vendor dir...

**Comment by hardys** [comment]
Actually it builds fine with the BUILD/TAGS defined in `05_build_ocp_installer.sh` so doing some local testing with that..

**Comment by stbenjam** [comment]
CI always fails on rebasing this way, it can't apply the patch.  The branch needs to be checked out.

Although, this is probably not worth testing at the moment - you won't get very far.  OpenShift moved to using api-int DNS address even for ignition, and we don't have that record.  There's work in progress, https://github.com/openshift-metal3/dev-scripts/pull/462, but that PR is not enough.  @celebdor and @cybertron are still working on a solution.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/633/

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 1]
```diff
@@ -172,8 +173,8 @@ then
 		--volume "$PWD:/assets:z" \
 		"${KUBE_APISERVER_OPERATOR_IMAGE}" \
 		/usr/bin/cluster-kube-apiserver-operator render \
-		--manifest-etcd-serving-ca=etcd-client-ca.crt \
-		--manifest-etcd-server-urls=$(etcd-server-urls $ETCD_MEMBERS) \
+		--manifest-etcd-serving-ca=etcd-ca-bundle.crt \
+		--manifest-etcd-server-urls={{.EtcdCluster}} \
```
Comment (markmc) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: This line shouldn't have changed - should still use etcd-server-urls function

**Review by markmc — CHANGES_REQUESTED** [review]

**Comment by markmc** [comment]
Another tiny change that could come with the rebase - https://github.com/markmc/kni-installer/commit/36b9ba614e5f25276deb61184b7fbcf775cf88d8

Fine with that going in post-merge though

**Comment by markmc** [comment]
Apart from those 2 small issues, I don't see any mechanical issues with the rebase. Nicely done!

**Comment by stbenjam** [comment]
I incorporated https://github.com/markmc/kni-installer/commit/36b9ba614e5f25276deb61184b7fbcf775cf88d8, and fixed the bootkube issue

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/637/

**Pull request merged**